pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+let timeout = null
+
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
+    setDataProviderUrl(newUrl)
+  }
+
+  useDebouncedEffect(() => {
+    if (dataProviderUrl === '') return
+
+    setIsLoading(true)
+
+    // TODO: Call API Here
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      if (dataProviderUrl === 'success.url')
+        setDiscoveredUrls([{ url: dataProviderUrl, isNew: true }])
+      else if (dataProviderUrl === 'duplicate.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }])
+      else if (dataProviderUrl === 'multiple-duplicates.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }, { url: dataProviderUrl }])
+      else setDiscoveredUrls([])
+
+      setIsLoading(false)
+    }, 1000)
+  }, [dataProviderUrl])
+
+  useEffect(() => {
+    setDataProviderUrl(defaultDataProviderUrl)
+  }, [defaultDataProviderUrl])
+
+  return [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ]
+}
+
+const getHelperMessage = (dataProviderUrl, discoveredUrls) => {
+  if (dataProviderUrl === '' || discoveredUrls === null) {
+    return {
+      message:
+        'It can be any URL: homepage address, any data resource address,' +
+        ' OAI-PMH endpoint or RIOXX endpoint',
```

Comment:
The messages should support rich text. The messages below should provide links to the existing data providers. Here the `OAI-PMH endpoint` should be a ling with the title _What is the OAI-PMH endpoint?_

Reply:
Can we keep this for next PR, we will need to add support for markdown I guess. It's related to https://github.com/oacore/search/issues/9

Reply:
Actually the new form components should cover it and we can put jsx elements there.